plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/ProgressWrapper.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.sdk.internal;
+
+import com.google.cloud.tools.managedcloudsdk.ProgressListener;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.SubMonitor;
+
+/**
+ * Wraps an {@link IProgressMonitor}. Assume that {@code totalWork} fits within an int which should
+ * be safe as it seems unlikely that {@code appengine-plugins-core} will download anything &gt; 2gb.
+ */
+public class ProgressWrapper implements ProgressListener {
+  private final IProgressMonitor wrapped;
+
+  private final SubMonitor progress;
+
+  public ProgressWrapper(IProgressMonitor progressMonitor) {
+    // Store wrapped monitor as SubMonitor#convert() says that caller is responsible for calling
+    // done() on argument
+    progress = SubMonitor.convert(progressMonitor);
```

Comment:
Hmm. I wonder if it's safe to use a SubMonitor directly, since it may trigger an OperationCancelledException when appengine-plugins-core is in operation.  What's the worst case?